### PR TITLE
feat: remove path parsing functionality

### LINKED ISSUE #1506: Preserved Catalog: do NOT parse paths
As of #1482 we decode writer ID, db name, table name, partition key, and chunk ID from the file paths. We should instead include them in the catalog as well as in the parquet file metadata (for #1376 ). The catalog should pass all this information to the `CatalogState` implementation, so the latter one should have no need to assume any path semantics.

# ACs:

- `CatalogState` implementation does not need to look at the path (it's just an opaque pointer)
- `Storage::parse_location` is gone
- the preserved catalog already does the conversion from "dirs and filename" to the store-native path (so the `CatalogState` does not need to do that)
- The path of the parquet file can be changed (with appropriate catalog pointer updates) and the contents will still be acceptable.

### PR DESCRIPTION
**Pull request by crepererum**
Closes #1506 .

### CODE REVIEW

File: parquet_file/src/chunk.rs [hunk 1]
```diff
@@ -105,24 +114,27 @@ pub struct Chunk {
 impl Chunk {
     /// Creates new chunk from given parquet metadata.
     pub fn new(
```
Comment (alamb) [inline parquet_file/src/chunk.rs:116]: this is a nice change to ensure the metadata and table_name / partition_key are consistent

File: parquet_file/src/storage.rs [hunk 2]
```diff
@@ -180,7 +180,8 @@ impl Storage {
     /// Return full path including filename in the object store to save a chunk
     /// table file.
     ///
-    /// See [`parse_location`](Self::parse_location) for parsing.
+    /// **Important: The resulting path should be treated as a black box. It might vary over time and is an
```
Comment (alamb) [inline parquet_file/src/storage.rs:183]: 👍  (fyi @NGA-TRAN )

File: server/src/db.rs [hunk 3]
```diff
@@ -1106,18 +1109,17 @@ impl CatalogState for Catalog {
     fn add(
         &self,
         object_store: Arc<ObjectStore>,
-        server_id: ServerId,
-        db_name: &str,
+        _server_id: ServerId,
```
Comment (alamb) [inline server/src/db.rs:1112]: perhaps (as a follow on PR) we could remove these arguments from the `CatalogState` trait as well

**Review by alamb — APPROVED** [review]
Looks like a nice cleanup to me

File: server/src/db.rs (hunk 3 above)
Comment (crepererum) [reply server/src/db.rs]: done

**Pull request merged**